File: internal/content/testdata/bench/java/nc-partition-equal-subset.java
import java.util.*;

class Solution {
    public boolean can_partition(long[] nums) {
        long total = 0;
        for (long n : nums) total += n;
        if (total % 2 != 0) return false;
        long target = total / 2;
        Set<Long> dp = new HashSet<>();
        dp.add(0L);
        for (long n : nums) {
            Set<Long> next = new HashSet<>(dp);
            for (long s : dp) {
                next.add(s + n);
            }
            dp = next;
            if (dp.contains(target)) return true;
        }
        return false;
    }
}

File: internal/content/testdata/bench/csharp/nc-partition-equal-subset.cs
using System.Collections.Generic;

public class Solution {
    public bool can_partition(long[] nums) {
        long total = 0;
        foreach (var n in nums) total += n;
        if (total % 2 != 0) return false;
        long target = total / 2;
        var dp = new HashSet<long> { 0 };
        foreach (var n in nums) {
            var next = new HashSet<long>(dp);
            foreach (var s in dp) next.Add(s + n);
            dp = next;
            if (dp.Contains(target)) return true;
        }
        return false;
    }
}
